scripts/retrieve_ci_failures.py

```diff
@@ -180,37 +191,49 @@ def get_fixed_by_commit_pushes():
 def retrieve_logs(fixed_by_commit_pushes, upload):
     os.makedirs(os.path.join("data", "ci_failures_logs"), exist_ok=True)
 
-    for push in tqdm(
-        fixed_by_commit_pushes.values(), total=len(fixed_by_commit_pushes)
-    ):
-        for failure in push["failures"]:
-            task_id = failure["task_id"]
-            retry_id = failure["retry_id"]
+    all_failures = [
+        failure
+        for push in fixed_by_commit_pushes.values()
+        for failure in push["failures"]
+    ]
 
-            log_path = os.path.join(
-                "data", "ci_failures_logs", f"{task_id}.{retry_id}.log"
-            )
-            log_zst_path = f"{log_path}.zst"
-            if os.path.exists(log_path) or os.path.exists(log_zst_path):
-                continue
+    with ThreadPoolExecutor(max_workers=10) as executor:
```

Comment:
How did you choose 10? Can we instead remove the parameter and rely on the default?